### ISSUE TITLE
How to fix the random seed when using REMBO, ALEBO, HeSBO

### ISSUE THREAD

**Issue by s974534426** [facebook/Ax#761]
- python3: 3.8.8
- ax version: 0.2.2
- os: ubuntu18.04

I'm using ```REMBOStrategy```, ```ALEBOStrategy```, ```HeSBOStrategy``` in ```ax.modelbridge.strategies```. I want to fix the random seed to make the code reproducible. I tried to pass ```random_seed``` parameter to ```optimize``` function, but it is useless. What should I do to make the code achieve the same results. The whole code is as follows:

``` python
import numpy as np
import random
import torch
from ax.utils.measurement.synthetic_functions import branin
from ax.modelbridge.strategies.alebo import ALEBOStrategy
from ax.service.managed_loop import optimize
from matplotlib import pyplot as plt


def branin_evaluation_function(parameterization):
    # Evaluates Branin on the first two parameters of the parameterization.
    # Other parameters are unused.
    x = np.array([parameterization["x0"], parameterization["x1"]])
    return {"objective": (branin(x), 0.0)}


random.seed(42)
np.random.seed(42)
torch.manual_seed(42)

parameters = [
    {"name": "x0", "type": "range", "bounds": [-5.0, 10.0], "value_type": "float"},
    {"name": "x1", "type": "range", "bounds": [0.0, 15.0], "value_type": "float"},
]
parameters.extend([
    {"name": f"x{i}", "type": "range", "bounds": [-5.0, 10.0], "value_type": "float"}
    for i in range(2, 100)
])
alebo_strategy = ALEBOStrategy(D=100, d=4, init_size=5)
best_parameters, values, experiment, model = optimize(
    parameters=parameters,
    experiment_name="branin_100",
    objective_name="objective",
    evaluation_function=branin_evaluation_function,
    minimize=True,
    total_trials=10,
    generation_strategy=alebo_strategy,
    random_seed=42,
)

objectives = np.array([trial.objective_mean for trial in experiment.trials.values()])

print(objectives[-1])
```

**Comment by lena-kashtelyan** [comment]
Let me loop in @bletham on this one!

**Comment by bletham** [comment]
Hi,
If you install the latest Ax version from github, then you can fix a random seed like this:
```Python
torch.manual_seed(42)
alebo_strategy = ALEBOStrategy(D=100, d=4, init_size=5, random_kwargs={"seed": 42})
```
Note that the seed must be specified in both places/ways. There are two big sources of randomness: the generation of the random embedding, and the selection of the initial random set of arms. The first specification (`torch.manual_seed`) will set the seed for generation of the random embedding, and so any `ALEBOStrategy` that you use after setting the torch seed will be working in the same embedding. In order to set the seed for the initial random set of arms, you have to use the `random_kwargs={"seed": 42}` argument to `ALEBOStrategy` (of course you can fill in whatever seed you'd like there).

Specifying the seed for the random initialization will only work if you install Ax from the latest master branch here on the github, since I had to make a change in https://github.com/facebook/Ax/commit/798fca5cdc2e5584874fb56dfc78108b935b0138 in order to get it working; on the current pypi release that kwarg will be filtered out and so won't do anything. You'll thus have seeded the random embedding but not the initial random arms. I will note however that I'd expect the random embedding to be a larger source of variance than the initial random arms.

You can verify that it is working by running this code (after the same code that you posted above for specifying the problem):

```Python
torch.manual_seed(42)
alebo_strategy = ALEBOStrategy(D=100, d=4, init_size=5, random_kwargs={"seed": 42})

best_parameters, values, experiment, model = optimize(
    parameters=parameters,
    experiment_name="branin_100",
    objective_name="objective",
    evaluation_function=branin_evaluation_function,
    minimize=True,
    total_trials=1,
    generation_strategy=alebo_strategy,
)
print(experiment.arms_by_name["0_0"])
```
When you run that repeatedly, you'll see that you are getting the same arm in the random initialization. (It will not be the same when you run `optimize` repeatedly without setting seeds).

**Comment by lena-kashtelyan** [comment]
Marking this as "fixready" since fix is now on latest main; will close this issue once next stable Ax version is released.

**Comment by lena-kashtelyan** [comment]
The fix for this is now part of the most recent stable release.
